source/common/filesystem/inotify/watcher_impl.cc

```diff
@@ -52,14 +52,18 @@ void WatcherImpl::addWatch(const std::string& path, uint32_t events, OnChangedCb
 
 void WatcherImpl::onInotifyEvent() {
   while (true) {
-    char buffer[sizeof(inotify_event) + NAME_MAX + 1];
+    uint8_t buffer[sizeof(inotify_event) + NAME_MAX + 1];
     ssize_t rc = read(inotify_fd_, &buffer, sizeof(buffer));
     if (rc == -1 && errno == EAGAIN) {
       return;
     }
+    if (rc < 0) {
```

Comment:
Should we just crash here?

Reply:
I added an `ASSERT`, so it'll fail loudly in dev/test and keep current behavior for opt.

It might be nice to have `ASSERT` log something on failure even in opt, so that bugs like this can be noticed if they ever occur.